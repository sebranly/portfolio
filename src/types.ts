// TODO: have a glossary on the website
// TODO: explain myProjects series
export enum Tag {
  All = 'All',
  ThreeDimensional = '3D',
  AIML = 'AIML',
  Archived = 'archived',
  C = 'C',
  CSS = 'CSS',
  CLI = 'CLI',
  DNS = 'DNS',
  GitHub = 'GitHub',
  GUI = 'GUI',
  HTML = 'HTML',
  i18n = 'i18n',
  JavaScript = 'JavaScript',
  Live = 'live',
  MyProjects = 'myProjects',
  NintendoDS = 'Nintendo DS',
  OpenGL = 'OpenGL',
  Puzzle = 'puzzle',
  Python = 'Python',
  React = 'React',
  SchoolProject = 'school project',
  SDL = 'SDL',
  VideoGame = 'video game',
  Website = 'website',
  PAlib = 'PAlib',
  PHP = 'PHP',
  SQL = 'SQL',
  TailwindCSS = 'TailwindCSS',
  TypeScript = 'TypeScript',
  Windows = 'Windows'
}

export enum Color {
  Black = 'black',
  White = 'white',
  Gray = 'gray',
  Red = 'red',
  Orange = 'orange',
  Amber = 'amber',
  Yellow = 'yellow',
  Green = 'green',
  Blue = 'blue',
  Indigo = 'indigo',
  Purple = 'purple',
  Pink = 'pink',
  Rose = 'rose'
}

export enum Role {
  Algorithm = 'Algorithm',
  Developer = 'Developer',
  Designer = 'Designer',
  ProductOwner = 'Product Owner',
  Tester = 'Tester'
}

export type Contributor = {
  female?: boolean;
  name: string;
  roles: Role[];
};

export type Project = {
  download?: string;
  title: string;
  contributors?: Contributor[];
  github?: string;
  subtitle: string;
  description: string[];
  tags: Tag[];
  video?: string;
  website?: string;
  years: number[];
  textualSnapshot?: string[];
};

export type Language = {
  shortName: string;
  name: string;
};
