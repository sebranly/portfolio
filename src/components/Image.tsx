import * as React from 'react';
import { Image as ImageType, ImageFolder } from '../types';
import { getImageLink } from '../utils/urls';
import { useTranslation } from 'react-i18next';

export interface ImageProps {
  className?: string;
  folder: ImageFolder;
  image: ImageType;
}

const Image: React.FC<ImageProps> = (props) => {
  const { className, folder, image } = props;

  const { t } = useTranslation();
  const { alt, extension, nb } = image;
  const altText = t(alt);
  const src = getImageLink(folder, nb, extension);

  return <img className={className} alt={altText} loading='lazy' src={src} />;
};

export { Image };
