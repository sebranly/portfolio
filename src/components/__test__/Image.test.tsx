import * as React from 'react';
import { render } from '@testing-library/react';

import { Image } from '../Image';
import { ImageExtension, ImageFolder } from '../../types';

test('Image', () => {
  const folder = ImageFolder.ConnectFour;
  const image = { alt: 'some alt', nb: 1, extension: ImageExtension.PNG };
  const createComponent = () => <Image folder={folder} image={image} />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
