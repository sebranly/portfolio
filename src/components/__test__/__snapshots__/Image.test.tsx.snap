// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image 1`] = `
NodeList [
  <img
    alt="some alt"
    src="https://raw.githubusercontent.com/sebranly/portfolio/main/public/img/projects/ConnectFour/img1.png"
  />,
]
`;
